djangocms_googlemap/models.py

```diff
@@ -1,90 +1,291 @@
-        help_text=_('Displayed under address in the bubble.'))
+MAP_TYPES = ['ROADMAP', 'SATELLITE', 'HYBRID', 'TERRAIN']
+MAP_TYPE_CHOICES = [(map_type, map_type) for map_type in MAP_TYPES]
+TRAVEL_MODES = ['DRIVING', 'BICYCLING', 'TRANSIT', 'WALKING']
+TRAVEL_MODE_CHOICES = [(travel_mode, travel_mode) for travel_mode in TRAVEL_MODES]
+ZOOM_LEVEL_CHOICES = map(lambda c: (c, str(c)), range(22)) # from 0 to 21
 
-    style = models.TextField(_("custom map style"), blank=True,
-        help_text=_('Provide a valid JSON configuration (escaped). See '
-            'developers.google.com/maps/documentation/javascript/styling'))
+CSS_WIDTH_RE = re.compile(r'^\d+(?:px|%)$')
+CSS_HEIGHT_RE = re.compile(r'^\d+px$')
 
-    ZOOM_LEVELS = map(lambda c: (c, str(c)), range(22))
-    zoom = models.PositiveSmallIntegerField(
-        _("zoom level"), choices=ZOOM_LEVELS, default=13)
 
-    lat = models.DecimalField(
-        _('latitude'), max_digits=10, decimal_places=6, null=True, blank=True,
-        help_text=_('Use latitude & longitude to fine tune the map position.'))
+# Add additional choices through the ``settings.py``.
+def get_templates():
+    choices = [
+        ('default', _('Default')),
+    ]
+    choices += getattr(
+        settings,
+        'DJANGOCMS_GOOGLEMAP_TEMPLATES',
+        [],
+    )
+    return choices
 
-    lng = models.DecimalField(
-        _('longitude'), max_digits=10, decimal_places=6, null=True, blank=True)
-
-    route_planer_title = models.CharField(
-        _("route planner title"), max_length=150, blank=True, null=True,
-        default=_('Calculate your fastest way to here'))
 
-    route_planer = models.BooleanField(_("route planner"), default=False)
+@python_2_unicode_compatible
+class GoogleMap(CMSPlugin):
+    """
+    Renders the Google Maps wrapper
+    """
+    translatable_content_excluded_fields = ['address', 'zipcode', 'width', 'height']
 
+    template = models.CharField(
+        verbose_name=_('Template'),
+        choices=get_templates(),
+        default=get_templates()[0][0],
+        max_length=255,
+    )
+    title = models.CharField(
+        verbose_name=_('Map title'),
+        max_length=255,
+        blank=True,
+    )
     width = models.CharField(
-        _('width'), max_length=6, default='100%',
-        help_text=_('Plugin width (in pixels or percent).'))
-
+        verbose_name=_('Width'),
+        max_length=6,
+        default='100%',
+        help_text=_('Width of the map (in pixels or percent, e.g. "100%" or "400px").'),
+    )
     height = models.CharField(
-        _('height'), max_length=6, default='400px',
-        help_text=_('Plugin height (in pixels).'))
-
-    info_window = models.BooleanField(
-        _('info window'), default=True,
-        help_text=_('Show textbox over marker'))
+        verbose_name=_('Height'),
+        max_length=6,
+        default='400px',
+        help_text=_('Height of the map (in pixels, e.g. "400px").'),
```

Comment:
so it can only be in pixels? not percentage? if so why not just make it a positiveintegerfield instead of charfield?